djinni/proxy_cache_impl.hpp

```diff
@@ -119,6 +162,9 @@ class ProxyCache<Traits>::Pimpl {
         }
     }
 
+    // Only ProxyCache<Traits>::get_base() can allocate these objects.
+    Pimpl() = default;
+
```

Comment:
We need this constructor `public` so that `SingletonHolder` can instantiate it; `SingletonHolder` is a managed class and, as such, afaik it can't be a "friend" of a native C++ type. Thus, the comment above no longer applies.

Reply:
I'm not sure if I can follow. So you recommend to remove the comment? 

```suggestion
    Pimpl() = default;

```